android/src/main/java/com/situm/plugin/PluginHelper.java

```diff
@@ -911,4 +917,34 @@ private void invokeCallback(Callback callback, Object args) {
         }
     }
 
+    public void onEnterGeofences(DeviceEventManagerModule.RCTDeviceEventEmitter eventEmitter) {
+        emitEnterGeofences = true;
+        createAndSetGeofenceListener(eventEmitter);
+    }
+
+    public void onExitGeofences(DeviceEventManagerModule.RCTDeviceEventEmitter eventEmitter) {
+        emitExitGeofences = true;
+        createAndSetGeofenceListener(eventEmitter);
+    }
+
+    private void createAndSetGeofenceListener(DeviceEventManagerModule.RCTDeviceEventEmitter eventEmitter) {
+        GeofenceListener geofenceListener = new GeofenceListener() {
+            public void onEnteredGeofences(List<Geofence> enteredGeofences) {
+                if (emitEnterGeofences) {
+                    emitGeofences(EVENT_ENTER_GEOFENCES, enteredGeofences);
+                }
+            }
+
+            public void onExitedGeofences(List<Geofence> exitedGeofences) {
+                if (emitExitGeofences) {
+                    emitGeofences(EVENT_EXIT_GEOFENCES, exitedGeofences);
+                }
+            }
+
+            private void emitGeofences(String event, List<Geofence> geofences) {
+                eventEmitter.emit(event, SitumMapper.mapList(geofences));
+            }
+        };
+        SitumSdk.locationManager().setGeofenceListener(geofenceListener);
```

Comment:
Non acabo de entender porque non se pode usar directamente o callback, porque hay que usar booleanos como flags e porque instanciar e settear o listener para cando entramos e para cando salimos

Reply:
O asunto é como sigue:

1. Non quero setear o listener no SDK a menos que sexa necesario, por iso instancio o listener só cando se invoca algún dos métodos.
2. O listener do SDK nativo é un para ambas callbacks (enter e exit), pero as callbacks RN van por separado para ser coherentes con outros plugin (ex. Capacitor, Flutter), por iso me aseguro de que se setea en ambos métodos.
3. Non me molesta que se instancie máis de 1 vez, é un listener moi sencillo.